DOC update the instructions on how to use Node.js (#2856)
pyodide/pyodide: docs/usage/index.md

## Node.js

```{note}
The following instructions have been tested with Node.js 18.5.0. To use

> Pyodide with older versions of node, you might need special command line
> instructions, see below.

Reviewer: ```suggestion
Pyodide with older versions of Node, you might need to use  additional command line
arguments, see below.
```

```

It is now possible to install the
[Pyodide npm package](https://www.npmjs.com/package/pyodide) in Node.js. To

> follow these instructions you need at least pyodide 0.21.0, which at the

Reviewer: ```suggestion
follow these instructions you need at least Pyodide 0.21.0, which at the
```

> follow these instructions you need at least pyodide 0.21.0, which at the
> moment of writing has not been released yet. You can explicity ask npm to use

Reviewer: ```suggestion
follow these instructions you need at least pyodide 0.21.0.
You can explicitly ask npm to use
```
 
Since the docs get published to the dev docs https://pyodide.org/en/latest/ and by the time it gets deployed to the stable docs which users use https://pyodide.org/en/stable, the release would happen.

the alpha version:

```
$ npm install "pyodide@>=0.21.0-alpha.2"
```

Once installed, you can run the following simple script:

```js
// hello_python.js
const { loadPyodide } = require("pyodide");

async function hello_python() {
  let pyodide = await loadPyodide();
  return pyodide.runPythonAsync("1+1");
}

hello_python().then((result) => {
  console.log("Python says that 1+1 =", result);
});
```

```
$ node hello_python.js
Loading distutils
Loaded distutils
Python initialization complete
Python says that 1+1= 2
```

Or you can use the REPL. To start the Node.js REPL with support for top level
await, use `node --experimental-repl-await`:

```
$ node --experimental-repl-await
Welcome to Node.js v18.5.0.
Type ".help" for more information.
> const { loadPyodide } = require("pyodide");
undefined
> let pyodide = await loadPyodide();
Loading distutils
Loaded distutils
Python initialization complete
undefined
> await pyodide.runPythonAsync("1+1");
2
```

> ### Older versions of Node.js

Reviewer: ```suggestion
### Node.js versions <0.17
```
to be a bit more specific

- `Node.js` versions 14.x and 16.x: to use certain features of Pyodide you
  need to manually install `node-fetch`, e.g. by doing `npm install node-fetch`.

- `Node.js v14.x`: you need to pass the option `--experimental-wasm-bigint`

>   when starting node. Note that this flag is not documented by `node --help`

Reviewer: ```suggestion
  when starting Node. Note that this flag is not documented by `node --help`
```
